### PR TITLE
RMB-471: Fix ddlgen/Index.java StringBuilder code smells

### PR DESCRIPTION
**Pull request by julianladisch**
* Remove unneccessary use of temporary StringBuilder
* Replace duplicate String constant by final variable
* Convert to guard clauses to avoid nesting/indentation
* Use line breaks for .append clauses for readability
* Reword comments
* Code formatting to comply with coding standards

### CODE REVIEW

File: dbschema/src/main/java/org/folio/rest/persist/ddlgen/Index.java [hunk 1]
```diff
@@ -76,41 +77,44 @@ public void setSqlExpression(String sqlExpression) {
...
-      for (int i = 0;i < splitIndex.length;i++) {
-        if(i != 0) {
-          result .append(" , ");
-        }
-        appendExpandedTerm(tableLoc, splitIndex[i],result);
+    String [] splitIndex = this.getMultiFieldNames().split(" *, *");
+
+    StringBuilder result = new StringBuilder("concat_space_sql(");
+    for (int i = 0; i < splitIndex.length; i++) {
+      if(i != 0) {
```
Comment (adamdickmeiss) [inline dbschema/src/main/java/org/folio/rest/persist/ddlgen/Index.java]: I do like a space between if and the (.. Just me being pedantic

File: dbschema/src/main/java/org/folio/rest/persist/ddlgen/Index.java [hunk 2]
```diff
@@ -76,41 +77,44 @@ public void setSqlExpression(String sqlExpression) {
...
+      result.append(table).append(".").append(Cql2PgUtil.cqlNameAsSqlText(JSONB, term));
       return;
     }
-    //case where the [*] is found
-    if(idx == term.length() - arrayToken.length()) {
-      expandedTerm.append(table).append(".").append(Cql2PgUtil.cqlNameAsSqlJson("jsonb",term.substring(0,idx)));
-      result.append("concat_array_object(").append(expandedTerm).append(")");
+    //case where [*] is found at the end
+    if(idx == term.length() - ARRAY_TOKEN.length()) {
+      result.append("concat_array_object(")
```
Comment (adamdickmeiss) [inline dbschema/src/main/java/org/folio/rest/persist/ddlgen/Index.java]: And here as well

File: dbschema/src/main/java/org/folio/rest/persist/ddlgen/Index.java (hunk 1 above)
Comment (julianladisch) [reply dbschema/src/main/java/org/folio/rest/persist/ddlgen/Index.java]: Fixed all code formatting issues.

File: dbschema/src/main/java/org/folio/rest/persist/ddlgen/Index.java (hunk 2 above)
Comment (julianladisch) [reply dbschema/src/main/java/org/folio/rest/persist/ddlgen/Index.java]: Fixed all code formatting issues.

**Pull request merged**